### DISCUSSION TITLE
Screen Edge Behavior (hiding/minimizing homebar)

### DISCUSSION THREAD

**Discussion by ecc521** [ionic-team/capacitor#6742] [Q&A]
On iPhones with the homebar (the little bar at the bottom of the screen used to close apps), there are methods to request the homebar be hidden when the user isn't touching the screen (prefersHomeIndicatorAutoHidden), as well as a method to disable user gestures near the edges of the screen, which when used on the bottom edge, also makes the homebar translucent (preferredScreenEdgesDeferringSystemGestures). 

I'm not seeing any Capacitor plugins that do this, and unfortunately, I'm unable to get cordova-plugin-screen-edges to work (https://github.com/distinctdan/cordova-plugin-screen-edges). 

I'm happy to write a plugin to do this behavior if there isn't anything working for Capacitor, but I don't have the skills with Swift & ObjC to actually write this without a ton of guessing. Suggestions?

**Comment by ecc521** [comment]
I was successful in directly modifying the capacitor files (CAPBridgeViewController.swift) using:

```
    override open var preferredScreenEdgesDeferringSystemGestures: UIRectEdge {
        return [.bottom] //Array of edges to defer
    }
```

Still not sure how to do this with a plugin but might be able to figure it out

**Comment by joepinion** [comment]
Seconded that ecc521's hack works. Seems like this could be a feature added to capacitor.config.json?

**Comment by ecc521** [comment]
Using the JSON config file is not optimal, as the screen edges locked should be able to be changed across different screens. An API is needed.
